x/observer/keeper/msg_server_vote_tss.go

```diff
@@ -42,11 +42,6 @@ func (k msgServer) VoteTSS(goCtx context.Context, msg *types.MsgVoteTSS) (*types
 		return &types.MsgVoteTSSResponse{}, errorsmod.Wrap(types.ErrKeygenNotFound, voteTSSid)
 	}
 
-	// Use a separate transaction to update keygen status to pending when trying to change the TSS address.
-	if keygen.Status == types.KeygenStatus_KeyGenSuccess {
-		return &types.MsgVoteTSSResponse{}, errorsmod.Wrap(types.ErrKeygenCompleted, voteTSSid)
-	}
-
```

Comment:
nit: Can you clarify why this is not needed anymore? Lacking context so I'd like to get this straight. :)

Reply:
We are removing this check , so as to allow observers to vote for ballots that have happened in the past and had to be discarded for some reason.
THe newly added checks below , would prevent such a vote from modifying the keygen status